channels/README.md

```diff
@@ -347,6 +347,25 @@ channel_settle |   |     open     | -----+
 ```
 
 
+## Contract execution
+
+Each participant of a state channels keeps a state tree representing the
+internal channel's state. This tree consists of accounts, contracts and
+contract calls. Although this tree is internal to the channel during a dispute
+resolution part of is posted on-chain in order to use the blockchain as an
+arbiter so despite being internal - the privacy could be further improved.
+
+Contract life cycle closesly represents the one on-chain: once a contract is
+created - it can be called by participants. Contract has state, balance and a
+list of calles. Participants can make calls and thus modify channel balance
+and state. Participants can inspect the return values of the past calls.
+Participants can remove a contract from a channel's state tree and
+redistribute its balance amongst them.
+
+If a dispute arises - a proof of inclusion of the contract inside channel's
+state tree is posted on-chain in order for the blockchain to enforce progress
+of the channel. That way we keep contracts in channels trustless.
```

Comment:
This paragraph could need some more details. How is progress enforced?

Reply:
That would be a subject to another PR - the one that delivers forcing progress on-chain